hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/CassStatement.java

```diff
@@ -312,9 +312,10 @@
 
         SELECT_TAGS_TRIGGERS_ALL_BY_NAME = "SELECT tenantId, triggers "
                 + "FROM " + keyspace + ".tags_triggers "
-                + "WHERE name = ? ";
+                + "WHERE name = ? "
+                + "ALLOW FILTERING";
```

Comment:
This "ALLOW FILTERING" may lead that something is "forcing" a non performance query.
It seems you need the tenantId.
Instead to force the driver for this query, this it can be done on client side, having a first query with tenants, and running multiple queries concurrently with the tenantId.
It can be interesting to think about it.
In my first version I introduced some ALLOW FILTERING or IN operators, but I could remove them in the 2nd version.